e2e/nomostest/testresourcegroup/resourcegroup.go

```diff
@@ -0,0 +1,150 @@
+func NewKptResourceGroup(nn types.NamespacedName, id string) *v1alpha1.ResourceGroup {
+	rg := &v1alpha1.ResourceGroup{}
+	rg.SetNamespace(nn.Namespace)
+	rg.SetName(nn.Name)
+	if id != "" {
+		rg.SetLabels(map[string]string{
+			common.InventoryLabel: id,
+		})
+	}
+	return rg
+}
+
+// AggregateResourceStatus is a helper function to generate an array of ResourceStatus
+// from ObjMetadata
+func AggregateResourceStatus(resources []v1alpha1.ObjMetadata, s v1alpha1.Status, sourceHash string) []v1alpha1.ResourceStatus {
+	resourceStatus := []v1alpha1.ResourceStatus{}
+	for _, resource := range resources {
+		resourceStatus = append(resourceStatus, v1alpha1.ResourceStatus{
+			ObjMetadata: resource,
+			Status:      s,
+			SourceHash:  sourceHash,
+		})
+	}
+	return resourceStatus
+}
+
+// InitialResourceStatus returns the initial/default ResourceStatus of a
+// ResourceGroup object.
+func InitialResourceStatus() v1alpha1.ResourceGroupStatus {
+	return v1alpha1.ResourceGroupStatus{
+		Conditions: []v1alpha1.Condition{
+			{
+				Type:    v1alpha1.Reconciling,
+				Status:  v1alpha1.FalseConditionStatus,
+				Reason:  resourcegroup.FinishReconciling,
+				Message: "finish reconciling",
+			},
+			{
+				Type:    v1alpha1.Stalled,
+				Status:  v1alpha1.FalseConditionStatus,
+				Reason:  resourcegroup.FinishReconciling,
+				Message: "finish reconciling",
+			},
+		},
+	}
+}
+
+// PatchResourceGroup modifies a ResourceGroup object on the cluster using
+// the provided mutateFn.
+func PatchResourceGroup(kubeClient *testkubeclient.KubeClient, nn types.NamespacedName, mutateFn func(group *v1alpha1.ResourceGroup)) error {
+	obj := &v1alpha1.ResourceGroup{}
+	if err := kubeClient.Get(nn.Name, nn.Namespace, obj); err != nil {
+		return err
+	}
+	obj2 := obj.DeepCopy()
+	mutateFn(obj2)
+	return kubeClient.Update(obj2)
+}
+
+// ApplyResources applies the list of resources to the cluster with the provided
+// ResourceGroup inventory ID.
+func ApplyResources(kubeClient *testkubeclient.KubeClient, resources []v1alpha1.ObjMetadata, id string) error {
+	for _, r := range resources {
+		u := &unstructured.Unstructured{}
+		u.SetName(r.Name)
+		u.SetNamespace(r.Namespace)
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   r.Group,
+			Version: "v1",
+			Kind:    r.Kind,
+		})
+		u.SetAnnotations(map[string]string{
+			"config.k8s.io/owning-inventory":      id,
+			resourcegroup.SourceHashAnnotationKey: "1234567890",
+		})
+
+		err := kubeClient.Get(r.Name, r.Namespace, u.DeepCopy())
+		if err == nil {
+			if updateErr := kubeClient.Update(u); updateErr != nil {
+				return updateErr
+			}
+		} else if apierrors.IsNotFound(err) {
+			if createErr := kubeClient.Create(u); createErr != nil {
+				return createErr
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// ControllerPodsNoRestart checks that no pods have restarted in the ResourceGroup
+// controller namespace.
+func ControllerPodsNoRestart(kubeClient *testkubeclient.KubeClient) error {
```

Comment:
This is only used in one place. I think you can inline it or make it a private local func near where it's used. Or maybe call it ListControllerPods and do the restart check in the test that calls it.

Reply:
I realized I missed a couple spots in the translated tests, it's used by three different test cases now

Reply:
If you want to keep it, it needs a verb in the name, like ValidateNoControllerPodRestarts maybe.

Reply:
Done.